### ISSUE TITLE
Errors with Final Version

### ISSUE THREAD

**Issue by 30suns** [ShMcK/ng2-socially-tutorial#3]
I wonder if you experience this same set of errors:

``` shell
git clone https://github.com/ShMcK/ng2-socially-tutorial/
cd ng2-socially-tutorial
meteor
[snip]
```

Then, load up the app, and click on any Party:

``` javascript
Uncaught ReferenceError: exports is not defined
parties-form.js:9 Uncaught ReferenceError: require is not defined
parties-list.js:9 Uncaught ReferenceError: require is not defined
party-details.js:12 Uncaught ReferenceError: require is not defined
app.js:9 Uncaught ReferenceError: require is not defined
angular2.dev.min.js:16 EXCEPTION: Parser Error: Unexpected token 'null' at column 22 in [party.name.toString()=$event] in null
angular2.dev.min.js:16 EXCEPTION: TypeError: Cannot read property 'elementBinders' of undefined
party-details.js:38 loading...
```

**Comment by ShMcK** [comment]
Yes, I have similar errors.

The first five `require`/`exports` errors are a result of the typescript-compiler. I hope that can be resolved in the near future. If you'd like, try `sclausen:angular2-typescript` as an alternative compiler, though it only works on the client and throws errors for typescript on the server.

The remaining errors I'm not sure about. 

Currently I'm focusing on making the tutorial more maintainable by referencing a real code base with a history that can be altered using rebase. I hope this can make the tutorial much easier to update, as changes can be permeated up through all steps.

Any help working out issues is always more than welcome!

**Comment by 30suns** [comment]
Good idea re: the codebase. Rather than by commit, maybe each step should reference branches (not tags because they can't be changed easily)? Just a thought.

**Comment by ShMcK** [comment]
Checkout this article on [maintainable tutorials](http://info.meteor.com/blog/step-by-step-tutorials-with-git). It can give you a better idea of how it all works.

You're also right though, users should also be able to access the tutorial by a branch pointing to a given step. 

Additionally, the tutorial needs unit tests to see what breaks and where.

**Comment by 30suns** [comment]
I've tried switching to `sclausen:angular2-typescript` but I get the same errors as before. In another tutorial I discovered there are some changes to how SystemJS works that requires a config section now. Not sure if that's relevant to this issue or not. There have also been a bunch of other updates to the libraries included in the meteor-angular2 package. I also kinda wonder why that package bundles its dependencies, wouldn't it be better to have a separate meteor package for each dependency (e.g., system.js, etc.).

**Comment by ShMcK** [comment]
You're absolutely right, the angular2-meteor and compiler need some work.

Actually, @sclausen's solution applies configuration settings to Typescript. However, this causes any Meteor backend TS files to fail. The settings are not compatible with both Meteor & Angular2. The work around is to only apply TS to the client, or use ES6.

Angular 2 currently requires a specific older version of System.js (0.16 rather than 0.18) to run. This will likely change at some point. 

Another [package](https://github.com/Urigo/Meteor-Angular2) is currently under development.
